lib/active_data.rb

```diff
@@ -59,6 +59,8 @@ def self.config
     case value
     when ::Hash then
       value
+    when defined?(ActionController::Parameters) && ActionController::Parameters
```

Comment:
I could define typecasters inside `if defined?(ActionController::Parameters)` to save `defined?` call on each typecast, but I'm not sure that we really care about it here.

Reply:
It's located before `::Hash` check because in rails-4.2 `ActionController::Parameters` inherits from `::Hash`.

Reply:
We probably do care about it, the less checks we have - the faster it is going to be, right?